opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+
+        Parameters:
+        compute_environment: str
+            optional, compute environment type to run model, if None, default using 'LOCAL_MACHINE'
+        num_machines: int
+            optional, number of machine to run model , if None, default using 1
+
+        Return:
+            None
+        """
+
+        if compute_environment is None or compute_environment == 0:
+            compute_environment = "LOCAL_MACHINE"
+        else:
+            subprocess.run("accelerate config")
+            return
+
+        if num_machines is None:
+            num_machines = 1
+
+        hf_cache_home = os.path.expanduser(
+            os.getenv(
+                "HF_HOME",
+                os.path.join(os.getenv("XDG_CACHE_HOME", "~/.cache"), "huggingface"),
+            )
+        )
+
+        cache_dir = os.path.join(hf_cache_home, "accelerate")
+
+        file_path = os.path.join(cache_dir + "/default_config.yaml")
+        print("generated config file: at" + file_path)
+        os.makedirs(os.path.dirname(file_path), exist_ok=True)
+        if num_processes is None:
+            if torch.cuda.is_available():
+                num_processes = torch.cuda.device_count()
+
+            else:
+                num_processes = 1
+
+        default_file = [
+            {
+                "compute_environment": compute_environment,
+                "deepspeed_config": {
+                    "gradient_accumulation_steps": 1,
+                    "offload_optimizer_device": "none",
+                    "offload_param_device": "none",
+                    "zero3_init_flag": False,
+                    "zero_stage": 2,
+                },
+                "distributed_type": "DEEPSPEED",
+                "downcast_bf16": "no",
+                "fsdp_config": {},
+                "machine_rank": 0,
+                "main_process_ip": None,
+                "main_process_port": None,
+                "main_training_function": "main",
+                "mixed_precision": "no",
+                "num_machines": num_machines,
+                "num_processes": num_processes,
+                "use_cpu": False,
+            }
+        ]
+        print(default_file)
+
+        with open(file_path, "w") as file:
+            yaml.dump(default_file, file)
+
+    @staticmethod
+    def qryrem(self, x):
+        # for removing the "QRY:" token
+        y = x.passages.split(" QRY:")[0].split("<|startoftext|>")
+        return y[1]
+
+    def is_notebook(self) -> bool:
+        try:
+            shell = get_ipython().__class__.__name__
+            # Jupyter notebook or qtconsole
+            if shell == "ZMQInteractiveShell":
+                return True
+                # Terminal running IPython
+            elif shell == "TerminalInteractiveShell":
+                return False
+            else:
+                return False
+        except NameError:
+            # Probably standard Python interpreter
+            return False
+
+
+if __name__ == "__main__":
```

Comment:
I'm not quite convinced why do we need this section in this class. I think that was making issue with circular dependency.

Reply:
Asking Sean, if we no longer need to run the training function on command line then I can remove the main function to take arguments from command line.

Reply:
Aligned to remove the main function to get parameters from command line, Considering for future development for commands